### PR TITLE
[37] notification update

### PR DESCRIPTION
**Pull request by steve1rm**
# Pull Request template

###  What does this PR solve?

- Creates the notification channel in the onCreate of the application class
- Changes the name to remindAtInMilliSecs
- Adds FLAG_IMMUATABLE
- Uses the DateTime instead of using the date and time separately

### CODE REVIEW

File: presentation/src/main/java/me/androidbox/presentation/BusbyTaskyApplication.kt [hunk 1]
```diff
@@ -1,7 +1,45 @@
...
+        return this.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+    }
+
+    private fun generateChannels(): Map<String, String> {
+        return mapOf(
+            "Notifications for EVENTS" to "event",
+            "Notifications for TASKS" to "task",
+            "Notifications for REMINDERS" to "reminders"
+        )
+    }
```
Comment (steve1rm) [inline presentation/src/main/java/me/androidbox/presentation/BusbyTaskyApplication.kt:44]: @philipplackner Is this the best way to create 3 separate channels

File: domain/src/main/java/me/androidbox/domain/alarm_manager/AlarmItem.kt [hunk 2]
```diff
@@ -15,7 +15,7 @@ fun Event.toAlarmItem(agendaType: AgendaType): AlarmItem {
         agendaId = this.id,
         title = this.title,
         description = this.description,
-        remindAt = this.remindAt * 1000,
+        remindAtInMilliSecs = this.remindAt * 1000,
```
Comment (steve1rm) [inline domain/src/main/java/me/androidbox/domain/alarm_manager/AlarmItem.kt]: @philipplackner Is it better to change the name here to something more readable.

 As the `setExact` needs it in milli seconds
```
alarmManager.setExact(
                AlarmManager.RTC_WAKEUP,
                alarmItem.remindAtInMilliSecs
               ......)
```

Or leave the name as it is `remindAt` and just do the following here
```
alarmManager.setExact(
                AlarmManager.RTC_WAKEUP,
                alarmItem.remindAt * 1000
               ......)
```

Comment (MKiperszmid) [reply domain/src/main/java/me/androidbox/domain/alarm_manager/AlarmItem.kt]: What I prefer is to have remindAt be the zoneddatetime, and then on the alarm manager convert it to milliseconds

Mostly so you always know you'll receive a valid date on the AlarmItem

File: presentation/src/main/java/me/androidbox/presentation/BusbyTaskyApplication.kt [hunk 3]
```diff
@@ -1,7 +1,45 @@
...
+        createNotificationChannel(generateChannels())
+    }
+
+    private fun createNotificationChannel(listOfChannel: Map<String, String>) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            listOfChannel.map { mapOfChannel ->
+                val notificationManager = getNotificationManager()
+                val notificationChannel = NotificationChannel(
+                    mapOfChannel.key,
+                    mapOfChannel.value,
```
Comment (MKiperszmid) [inline presentation/src/main/java/me/androidbox/presentation/BusbyTaskyApplication.kt:25]: This is correct

But
When you check the key and values on the map, they are the other way wrong

Key should be "event", "reminder", "task", and value should have the description
But on the map creation, you are doing it the opposite way

File: presentation/src/main/java/me/androidbox/presentation/BusbyTaskyApplication.kt (hunk 1 above)
Comment (MKiperszmid) [reply presentation/src/main/java/me/androidbox/presentation/BusbyTaskyApplication.kt:44]: Here, read comment above

**Review by MKiperszmid — APPROVED** [review]

File: domain/src/main/java/me/androidbox/domain/alarm_manager/AlarmItem.kt (hunk 2 above)
Comment (steve1rm) [reply domain/src/main/java/me/androidbox/domain/alarm_manager/AlarmItem.kt]: @philipplackner what do you think. I do perfer @MKiperszmid as in my second suggestion. And convert it to millis seconds only when needed.

File: presentation/src/main/java/me/androidbox/presentation/BusbyTaskyApplication.kt (hunk 3 above)
Comment (steve1rm) [reply presentation/src/main/java/me/androidbox/presentation/BusbyTaskyApplication.kt:25]: @MKiperszmid Thanks for catching that. I have updated

**Comment by steve1rm** [comment]
@MKiperszmid @philipplackner Decided to go with @MKiperszmid suggestion about converting to millis seconds only when its needed.

**Pull request merged**